editor/components/autosave-monitor/index.js

```diff
@@ -35,13 +38,21 @@ export class AutosaveMonitor extends Component {
 
 export default compose( [
 	withSelect( ( select ) => {
-		const { isEditedPostDirty, isEditedPostSaveable } = select( 'core/editor' );
+		const {
+			isEditedPostDirty,
+			isEditedPostSaveable,
+			isPostAutosaveable,
+			getEditorSettings,
+		} = select( 'core/editor' );
 		return {
 			isDirty: isEditedPostDirty(),
 			isSaveable: isEditedPostSaveable(),
+			isAutosaveable: isPostAutosaveable(),
+			editorSettings: getEditorSettings(),
```

Comment:
Minor: We should only need to pass in the specific value being used: `autosaveInterval`, for a few reasons:

- Tend to prefer simpler prop value types, as complex types (`{}` and `[]`) are prone to incurring more reconciliation than intended / necessary by strict inequality, where `{} !== {}` but `10 === 10`. 
   - I checked, and this isn't a concern here with the current implementation of `getEditorSettings`
- In the strictest sense of a presentational component, `AutosaveMonitor` doesn't care about settings as a whole, it cares specifically about an interval value, and the props passed in via `withSelect` should reflect these requirements.

Aside: The desire for a singular value to be passed highlights that it may be wise for a new singular `getEditorSetting` selector to be created, or even replacing the `getEditorSettings` function to avoid it being a source of future maintainability headaches (i.e. footgun).

Reply:
Good points and agrees with my sense that something could be better here. changed in https://github.com/WordPress/gutenberg/pull/6257/commits/2012003b091d4a6201b6208f33568e8842d4c2d6